src/libraries/System.Private.CoreLib/src/System/IO/StreamReader.cs

```diff
@@ -245,29 +245,90 @@ protected override void Dispose(bool disposing)
             {
                 return;
             }
-            _disposed = true;
```

Comment:
After reading https://github.com/dotnet/runtime/pull/89477#discussion_r1276760489, I'm not sure if we want to do this for StreamReader, seems like StreamWriter has a bad implementation of Dispose[Async] where it will never signal _dispose = true if _closable = false (i.e: if leaveOpen = true was specified).

Reply:
In below snippet I was expecting an exception to be thrown but instead the StreamWriter keeps working:
```cs
var ms = new MemoryStream();
var sw = new StreamWriter(ms, leaveOpen: true);

sw.Write("Hello");
sw.Dispose();

sw.Write(" World");
sw.Flush();
Console.WriteLine(Encoding.UTF8.GetString(ms.ToArray())); // prints "Hello World".
```
If I change leaveOpen to false, it does throw as I expected.

I suggest we keep setting _disposed here and we should discuss the StreamWriter bug in a separate issue.

Reply:
Ok, I will change this for the Reader to do the if closeable inside the try-catch-block